outputs/client.go

```diff
@@ -112,10 +120,32 @@ func (c *Client) Post(payload interface{}) error {
 
 	customTransport := http.DefaultTransport.(*http.Transport).Clone()
 
-	if c.Config.CheckCert == false {
-		// #nosec G402 This is only set as a result of explicit configuration
+	if c.MutualTLSEnabled {
+		// Load client cert
+		cert, err := tls.LoadX509KeyPair(c.Config.MutualTLSFilesPath+MutualTLSClientCertFilename, c.Config.MutualTLSFilesPath+MutualTLSClientKeyFilename)
+		if err != nil {
+			log.Printf("[ERROR] : %v - %v\n", c.OutputType, err.Error())
+		}
+
+		// Load CA cert
+		caCert, err := ioutil.ReadFile(c.Config.MutualTLSFilesPath + MutualTLSCacertFilename)
+		if err != nil {
+			log.Printf("[ERROR] : %v - %v\n", c.OutputType, err.Error())
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
 		customTransport.TLSClientConfig = &tls.Config{
-			InsecureSkipVerify: true,
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      caCertPool,
+			MinVersion:   tls.VersionTLS12,
+		}
+	} else {
+		// With MutualTLS enabled, the check cert flag is ignored
+		if c.Config.CheckCert == false {
```

Comment:
If we add mTLS setting by output, maybe this parameter should not be global either. wdyt?

Reply:
Agree, I'll make it configurable per output. Are you ok with the current behaviour? (i.e: if the output has mutualtls=true, a checkcert=false flag will be ignored)

Reply:
Totally makes sense.